test/unittest/blas1/blas1_scal_test.cpp

```diff
@@ -30,33 +30,53 @@ using combination_t = std::tuple<std::string, int, scalar_t, int>;
 
 template <typename scalar_t, helper::AllocType mem_alloc>
 void run_test(const combination_t<scalar_t> combi) {
+  using ref_scalar_t = typename utils::ReferenceType<scalar_t>::type;
   std::string alloc;
   index_t size;
   scalar_t alpha;
   index_t incX;
   std::tie(alloc, size, alpha, incX) = combi;
 
+  index_t size_x = size * incX;
+
   // Input/output vector
-  std::vector<scalar_t> x_v(size * incX);
-  std::vector<scalar_t> x_cpu_v(x_v);
+  std::vector<scalar_t> x_v(size_x);
+  std::vector<ref_scalar_t> x_cpu_v(size_x);
+
+  auto q = make_queue();
 
-  // Reference implementation
-  reference_blas::scal(size, alpha, x_cpu_v.data(), incX);
+  constexpr const bool is_sycl_half = std::is_same_v<scalar_t, cl::sycl::half>;
+  if (is_sycl_half && !q.get_device().has(cl::sycl::aspect::fp16)) {
+    GTEST_SKIP() << "Unsupported fp16 (half) on this device.";
+  }
+
+  if constexpr (is_sycl_half) {
+    // Float-type variables for reference ops
+    ref_scalar_t alpha_f = alpha;
+    // sycl::half to float reference type
+    std::transform(x_v.begin(), x_v.end(), x_cpu_v.begin(),
+                   [](scalar_t x) { return (static_cast<ref_scalar_t>(x)); });
+    // Reference implementation
+    reference_blas::scal(size, alpha_f, x_cpu_v.data(), incX);
```

Comment:
Did you considered performing the type conversion within the `reference_blas` implementation rather than re-implementing every time for each unittest/benchmark? E.g.,
```
 -- common/include/common/system_reference_blas.hpp --
 template <typename scalar_t>
 void scal(const int n, const scalar_t alpha, scalar_t x[], const int incX) {
  if constexpr (!std::is_same_v<scalar_t, cl::sycl::half>) {
    auto func = blas_system_function<scalar_t>(&cblas_sscal, &cblas_dscal);
    func(n, alpha, x, incX);
  } else {
    std::vector<float> x_f( n * incX);
    for(int i = 0; i < n * incX; ++i)
      x_f[i] = static_cast<float>(x[i]);
    cblas_sscal(n, float(alpha), x_f.data(), incX);
    for(int i = 0; i < n * incX; ++i)
      x[i] = static_cast<cl::sycl::half>(x_f[i]);
  }
 }
```
This kind of solution will allow to simplify all the unittests/benchmarks of the operators supporting halfs. Moreover, this will remove the need for a specific `compare_vectors` function.